src/api/goods.ts

```
@@ -4,13 +4,29 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL).then(response => {
+    if (!response.ok) {
+      throw new Error('Network response was not ok');
+    }
+
+    return response.json();
+  });
 }
 
 export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+  return getAll()
+    .then(goods =>
+      goods.sort((a, b) => a.name.localeCompare(b.name)).slice(0, 5),
+    )
+    .catch(error => {
+      throw new Error(error.message);
+    });
 };
 
 export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+  return getAll()
+    .then(goods => goods.filter(good => good.color === 'red'))
+    .catch(error => {
+      throw new Error(error.message);
```

To provide more informative error messages, you should throw a new error with a more descriptive message instead of just passing the original error message. This will help in debugging.

```javascript
.catch(error => {
  throw new Error(`Failed to get red goods: ${error.message}`);
});
```

This aligns with the principle of providing informative error messages to aid in debugging.